src/main/java/org/apache/commons/lang3/text/StrSubstitutor.java

```diff
@@ -657,11 +664,12 @@ public boolean replaceIn(final StringBuffer source, final int offset, final int
      * @return true if altered
      * @since 3.2
      */
+    @SuppressWarnings("index:argument.type.incompatible") // #1: source.length() is as required @LTLengthOf(value="source", offset="-1"), can't find the length function in the stub file
```

Comment:
I don't understand the problem here

Reply:
In the stub file of StringBuilder that I attained by the following command:
java -cp $CHECKERFRAMEWORK/checker/dist/checker.jar org.checkerframework.framework.stub.StubGenerator java.lang.StringBuilder > StringBuilder.astub

I can't find the length() function to annotate.
The following is the stub file:
```
package java.lang;

class StringBuilder extends AbstractStringBuilder implements Serializable, CharSequence {
    StringBuilder();
    StringBuilder(int arg0);
    StringBuilder(String arg0);
    StringBuilder(CharSequence arg0);
    StringBuilder append(Object arg0);
    StringBuilder append(String arg0);
    StringBuilder append(StringBuffer arg0);
    StringBuilder append(CharSequence arg0);
    StringBuilder append(CharSequence arg0, int arg1, int arg2);
    StringBuilder append(char[] arg0);
    StringBuilder append(char[] arg0, int arg1, int arg2);
    StringBuilder append(boolean arg0);
    StringBuilder append(char arg0);
    StringBuilder append(int arg0);
    StringBuilder append(long arg0);
    StringBuilder append(float arg0);
    StringBuilder append(double arg0);
    StringBuilder appendCodePoint(int arg0);
    StringBuilder delete(int arg0, int arg1);
    StringBuilder deleteCharAt(int arg0);
    StringBuilder replace(int arg0, int arg1, String arg2);
    StringBuilder insert(int arg0, char[] arg1, int arg2, int arg3);
    StringBuilder insert(int arg0, Object arg1);
    StringBuilder insert(int arg0, String arg1);
    StringBuilder insert(int arg0, char[] arg1);
    StringBuilder insert(int arg0, CharSequence arg1);
    StringBuilder insert(int arg0, CharSequence arg1, int arg2, int arg3);
    StringBuilder insert(int arg0, boolean arg1);
    StringBuilder insert(int arg0, char arg1);
    StringBuilder insert(int arg0, int arg1);
    StringBuilder insert(int arg0, long arg1);
    StringBuilder insert(int arg0, float arg1);
    StringBuilder insert(int arg0, double arg1);
    int indexOf(String arg0);
    int indexOf(String arg0, int arg1);
    int lastIndexOf(String arg0);
    int lastIndexOf(String arg0, int arg1);
    StringBuilder reverse();
    String toString();
}

```

Reply:
There are two issues here:
1. `StringBuilder` is a mutable sequence, so upperbound annotations cannot apply to it in general (since its length might change, making the annotations invalid)
2. `StringBuilder` doesn't define its own `length` function; instead, it inherits from `CharSequence`

I would suppress this warning by noting that `StringBuilder` is mutable, so you can't be sure of this in general.